components/mediation-ui/org.wso2.carbon.message.processor.ui/src/main/java/org/wso2/carbon/message/processor/ui/MessageProcessorAdminServiceClient.java

```diff
@@ -452,15 +452,18 @@ public String browseMessage(String processorName) throws Exception {
      * Pop the message from the associated queue
      *
      * @param processorName The MessageProcessor name
+     * @return <code>true</code> if popMessage is successful, else <code>false</code>
      * @throws Exception
      */
-    public void popMessage(String processorName) throws Exception {
+    public boolean popMessage(String processorName) throws Exception {
         try {
             if (processorName != null) {
```

Comment:
Can the processor name be null?

Reply:
No its not possible. But I was asked to add a null check just in case. I think there was a concern that the functions could be accessed externally or something along those lines. 


Shall I remove the null check?

Reply:
Even if the service is externally accessed and a null value is passed in, we are supposed to return an exception rather than returning false, giving a wrong impression that a message does not exist rather than saying that the message processor name is invalid. Therefore, I suggest we remove the null check. WDYT?

Reply:
Yes, thats is true. 

returning false while there is a message in the queue just because we called the function without a processor name doesn't make sense. 

ill remove the null check